### PR TITLE
Refactor: Move functions in linters to top-level

### PR DESCRIPTION
**Pull request by queengooborg**
This PR moves all functions out of specific scopes and into top-level for legibility purposes.

### CODE REVIEW

**Review by alshamiri5 — APPROVED** [review]

File: test/linter/test-versions.js [hunk 1]
```diff
@@ -147,6 +147,22 @@ function checkVersions(supportData, relPath, logger) {
   }
 }
 
+/**
+ * @param {Identifier} data
+ * @param {string} [relPath]
+ */
+function findSupport(data, logger, relPath) {
```
Comment (caugner) [inline test/linter/test-versions.js:154]: 1. logger param is missing in JSDoc
2. Adding `@returns {void}` would also make sense. 🤓 
3. Not sure why `relPath` is in brackets. 🤷

File: test/linter/test-real-values.js [hunk 2]
```diff
@@ -76,6 +76,33 @@ function checkRealValues(supportData, blockList, relPath, logger) {
...
+        logger,
+        relPath ? `${relPath}.${prop}` : `${prop}`,
+      );
+    }
+  }
+}
+
 /**
  * @param {string} filename
  */
```
Comment (caugner) [inline test/linter/test-real-values.js:108]: Wouldn't hurt to add a `@returns` annotation at the same time.

```suggestion
 * @returns {boolean}
 */
```

File: test/linter/test-versions.js [hunk 3]
```diff
@@ -147,6 +147,22 @@ function checkVersions(supportData, relPath, logger) {
...
+    const sub = data[prop];
+    if (typeof sub === 'object') {
+      findSupport(sub, logger, relPath ? `${relPath}.${prop}` : `${prop}`);
+    }
+  }
+}
+
 /**
  * @param {string} filename
  */
```
Comment (caugner) [inline test/linter/test-versions.js:168]:
```suggestion
 * @returns {boolean}
 */
```

File: test/linter/test-real-values.js [hunk 4]
```diff
@@ -76,6 +76,33 @@ function checkRealValues(supportData, blockList, relPath, logger) {
   }
 }
 
+/**
+ * @param {Identifier} data
+ * @param {string} [relPath]
+ */
+function findSupport(data, category, logger, relPath) {
```
Comment (caugner) [inline test/linter/test-real-values.js:83]: Missing JSDoc entry for the new logger param.

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM, just missing JSDoc for new `logger` parameter

**Pull request merged**